homeassistant/components/lg_thinq/entity.py

```diff
@@ -29,43 +33,42 @@ def __init__(
         self,
         coordinator: DeviceDataUpdateCoordinator,
         entity_description: EntityDescription,
-        property: ThinQProperty,
+        idx: str,
     ) -> None:
         """Initialize an entity."""
         super().__init__(coordinator)
 
         self.entity_description = entity_description
-        self.property = property
+        self.idx = idx
+
         self._attr_device_info = dr.DeviceInfo(
             identifiers={(DOMAIN, coordinator.unique_id)},
             manufacturer=COMPANY,
-            model=coordinator.device_api.model_name,
+            model=coordinator.api.device.model_name,
             name=coordinator.device_name,
         )
-
-        # Set the unique key. If there exist a location, add the prefix location name.
-        unique_key = (
-            f"{entity_description.key}"
-            if property.location is None
-            else f"{property.location}_{entity_description.key}"
-        )
-        self._attr_unique_id = f"{coordinator.unique_id}_{unique_key}"
+        self._attr_unique_id = f"{coordinator.unique_id}_{self.idx}"
+        self._attr_translation_placeholders = {
+            "location": (
+                ""
+                if self.location is None
+                or self.location in (Location.MAIN, Location.OVEN, coordinator.sub_id)
+                else f"{self.location} "
+            )
+        }
 
         # Update initial status.
         self._update_status()
 
-    async def async_post_value(self, value: Any) -> None:
-        """Post the value of entity to server."""
-        try:
-            await self.property.async_post_value(value)
-        except ThinQAPIException as exc:
-            raise ServiceValidationError(
-                exc.message,
-                translation_domain=DOMAIN,
-                translation_key=exc.code,
-            ) from exc
-        finally:
-            await self.coordinator.async_request_refresh()
+    @property
+    def data(self) -> PropertyState:
+        """Return the state data of entity."""
+        return self.coordinator.data.get(self.idx, EMPTY_STATE)
+
+    @property
+    def location(self) -> str | None:
+        """Return the location of entity."""
+        return self.coordinator.api.get_location_for_idx(self.idx)
```

Comment:
Does this change on runtime?

Reply:
'location' is static.
And it is handled(stored) in the library.

Reply:
Should we set it in the constructor? Or is there a reason no to do it?

Reply:
Renamed.

Reply:
Moved into the constructor.